core/shared/src/main/scala/fs2/pipe.scala

```diff
@@ -135,6 +136,28 @@ object pipe {
     }
   }
 
+  /** Debounce the stream with a minimum delay of `d` between each element */
+  def debounce[F[_], I](d: FiniteDuration)(implicit F: Effect[F], scheduler: Scheduler, ec: ExecutionContext): Pipe[F, I, I] = {
+    def go(i: I, h1: Handle[F, I]): Pull[F, I, Nothing] = {
+      time.sleep[F](d).open.flatMap { h2 =>
+        h2.awaitAsync.flatMap { l =>
+          h1.awaitAsync.flatMap { r =>
+            (l race r).pull.flatMap {
+              case Left(_) => Pull.output1(i) >> r.pull.flatMap(identity).flatMap {
+                case (hd, tl) => go(hd.last, tl)
+              }
+              case Right(r) => r.optional.flatMap {
+                case Some((hd, tl)) => go(hd.last, tl)
+                case None => l.pull.flatMap(identity).flatMap { _ => Pull.output1(i) >> Pull.done }
```

Comment:
I'm not entirely certain whether this should be emitted straight away or after the `time.sleep`.